proto/frontend/src/device_mgr.cpp

```diff
@@ -999,27 +1014,18 @@ class DeviceMgrImp {
   Status direct_counter_read(const p4::DirectCounterEntry &counter_entry,
                              const SessionTemp &session,
                              p4::ReadResponse *response) const {
-    auto counter_id = counter_entry.counter_id();
-    if (counter_id == 0 && counter_entry.has_table_entry()) {
-      RETURN_ERROR_STATUS(
-          Code::INVALID_ARGUMENT,
-          "When reading direct counters, you cannot use a counter_id of zero "
-          "with a non-empty table_entry");
+    if (!counter_entry.has_table_entry()) {
+      RETURN_ERROR_STATUS(Code::UNIMPLEMENTED,
+                          "Reading ALL direct counters is not supported yet");
     }
-    if (counter_id == 0) {
+    const auto &table_entry = counter_entry.table_entry();
+    if (table_entry.table_id() == 0) {
       RETURN_ERROR_STATUS(Code::UNIMPLEMENTED,
```

Comment:
same as above, should we have 2 ways of doing this?

Reply:
Fixed